### ISSUE TITLE
Have VS Code's send auto-completion request on subsequent typed characters

### ISSUE THREAD

**Issue by michelleangela** [microsoft/vscode-cpptools#9036]
Bug type: Language Service

**Describe the bug**
- OS: Windows
- VS Code Version: 1.9.4

**Steps to reproduce**

1. type `ao` and wait for auto-completion list to appear

**Expected behavior**
Reserved identifiers such as ` __ARRAY_OPERATORS__ ` should be filtered out.

**Cause**
When typing "a" the language server sends VS Code all the "a" results with inexact matches, and when typing the "o" VS Code doesn't resend another completion request to language server and does it's own "fuzzy" matching which results in having no filtering.

**Proposed solution**
Force VS Code send another auto-completion request.

**Comment by sean-mcmanus** [comment]
Yeah, I think the completion result has an "incomplete" field that could be set to true. However, it's possible VS Code may not actually check that in this particular case (not sure). Also, it's possible there could be a performance issue with sending an IntelliSense completion request for every character typed (i.e. it could create a big queue of completion requests that could cause no completions to appear). Another approach may be to not return "inexact matches" for the filtered out cases.

**Comment by bobbrow** [comment]
Perhaps we could send the "incomplete" flag if there are no exact matches.  I'm not sure what the implications would be so we'd need to test it, but I think that would address this case.
